app/code/Magento/Rule/Model/Condition/Product/AbstractProduct.php

```diff
@@ -657,23 +658,11 @@ public function validateByEntityId($productId)
      * Retrieve category ids where product is available
      *
      * @param int $productId
-     * @return array
```

Comment:
@lbajsarowicz can we move it to separate PR in order not to wait for approval? This PR brings good performance improvement!

Reply:
Yes, we can 🤣